onnxruntime/core/mlas/lib/platform.cpp

```diff
@@ -425,3 +434,58 @@ Return Value:
     return MLAS_DEFAULT_PREFERRED_BUFFER_ALIGNMENT;
 #endif
 }
+
+#if defined(MLAS_TARGET_ARM64) && defined(__linux__)
+static MlasCoreType* mlas_coretype_tbl = nullptr;
+static uint32_t mlas_coretype_tbl_size = 0;
+static std::once_flag mlas_init_coretype_tbl;
+#endif
+
+MlasCoreType
+MlasGetCoreType()
+{
+
+#if defined(MLAS_TARGET_ARM64) && defined(__linux__)
+    std::call_once(mlas_init_coretype_tbl, []() {
+        mlas_coretype_tbl_size = std::thread::hardware_concurrency();
+        if (mlas_coretype_tbl_size == 0) {
+            return;
+        }
+        mlas_coretype_tbl = (MlasCoreType*)malloc(sizeof(MlasCoreType) * mlas_coretype_tbl_size);
```

Comment:
>mlas_coretype_tbl [](http://example.com/codeflow?start=8&length=17)

a memory leak. it is harmless though.

Reply:
do we have a hook to destruct MLAS?

Reply:
How about std::unique_ptr? If you move it to CPUInfo, it can be handled in the constructor and deconstructor.